lib/controller.dart

```diff
@@ -189,332 +218,454 @@ void loadGeoMapsData() async {
+      case model.ChartType.map:
+        var colors = chart.colors ?? chart_helper.chartDefaultColors;
+        var computedChart = model.ComputedMapChart(
+            chart.data_path,
+            chart.title,
+            chart.narrative,
+            colors,
+            chart.fields.values,
+            chart.fields.values.map((_) => [0.0, 0.0]).toList(),
+            chart.fields.values.map((_) => [0.0, 0.0]).toList(),
+            ['', ''],
+            [chart.geography.country, chart.geography.regionLevel.name]);
+        _computedCharts.add(computedChart);
         break;
       case model.ChartType.time_series:
-        view.renderChart(
+        var buckets = <DateTime, List<num>>{};
+        var computedChart = model.ComputedTimeSeriesChart(
+            chart.data_path,
+            chart.doc_name,
             chart.title,
             chart.narrative,
-            chart_helper.generateTimeSeriesChartConfig(
-                chart, _dataNormalisationEnabled, _stackTimeSeriesEnabled));
+            chart.colors,
+            chart.data_label,
+            chart.fields.labels,
+            buckets);
+        _computedCharts.add(computedChart);
         break;
-      case model.ChartType.map:
-        var mapData =
-            _mapsGeoJSON[chart.geography.country][chart.geography.regionLevel];
-        var mapValues = Map<String, List<num>>();
-        var mapComparisonValues = Map<String, List<num>>();
-        for (var field in chart.fields) {
-          var regionName = field.field.value.toString();
-          var normalisationValue =
-              _dataNormalisationEnabled ? 100 : _filterValuesCount;
-          var comparisonNormalisationValue =
-              _dataNormalisationEnabled ? 100 : _comparisonFilterValuesCount;
-
-          mapValues[regionName] = [
-            field.bucket[0],
-            field.bucket[0] / normalisationValue,
-          ];
-          mapComparisonValues[regionName] = [
-            field.bucket[1],
-            field.bucket[1] / comparisonNormalisationValue
-          ];
-        }
-
-        view.renderGeoMap(
-          chart.title,
-          chart.narrative,
-          mapData,
-          mapValues,
-          mapComparisonValues,
-          _dataComparisonEnabled,
-          _dataNormalisationEnabled,
-          chart.colors ?? chart_helper.barChartDefaultColors,
-        );
+      case model.ChartType.funnel:
+        var computedChart = model.ComputedFunnelChart(
+            chart.data_path,
+            chart.title,
+            chart.narrative,
+            chart.colors ?? chart_helper.chartDefaultColors,
+            [],
+            [],
+            chart.is_paired);
+        _computedCharts.add(computedChart);
         break;
       default:
-        logger.error('No such chart type ${chart.type}');
-        view.showAlert(
-            'Warning: Chart type ${chart.type} listed in your config is not supported.');
+        _computedCharts.add(null);
+        logger.error(
```

Comment:
I don't think it makes sense to add a null chart, does it? you'll just end up needing to always check if a computed chart is null. I think easier to just show the error in the UI, and then ignore it.

Reply:
I just added a null so I could match the charts in the config and the computed ones by index. Or I could add an ID to the chart at the time of computation. Will update it